plugins/modules/api_gateway.py

```diff
@@ -98,17 +98,31 @@
     choices: ['EDGE', 'REGIONAL', 'PRIVATE']
     type: str
     default: EDGE
+  name:
+    description:
+      - The name of the RestApi.
+    type: str
+    version_added: 6.2.0
+  lookup:
+    description:
+      - Look up API gateway by either I(tags) (and I(name) if supplied) or by I(api_id).
+      - If I(lookup=tag) and I(tags) is not specified then no lookup for an existing API gateway
+        is performed and a new API gateway will be created.
+      - When using I(lookup=tag), multiple matches being found will result in a failure and no changes will be made.
+      - To change the tags of a API gateway use I(lookup=id).
+    default: tag
+    choices: [ 'tag', 'id' ]
+    type: str
+    version_added: 6.2.0
 author:
   - 'Michael De La Rue (@mikedlr)'
 notes:
-  - A future version of this module will probably use tags or another
-    ID so that an API can be created only once.
-  - As an early work around an intermediate version will probably do
-    the same using a tag embedded in the API name.
+  - 'Tags are used to uniquely identify API gateway when the I(api_id) is not supplied. version_added=6.2.0'
 extends_documentation_fragment:
   - amazon.aws.common.modules
   - amazon.aws.region.modules
   - amazon.aws.boto3
+  - amazon.aws.tags
```

Comment:
Can you also please check if the RETURN block is up to date with the latest features?

Reply:
@alinabuzachis the RETURN block is still valid since every API gateway information is inside the `configure_response`